usi/src/main/java/daq/usi/UsiImpl.java

```diff
@@ -20,36 +20,37 @@ public UsiImpl() {
     switchControllers = new HashMap<>();
   }
 
+  private SwitchController createController(SwitchInfo switchInfo) {
+    SwitchController newController;
+    switch (switchInfo.getModel()) {
+      case ALLIED_TELESIS_X230: {
+        newController =
+            new AlliedTelesisX230(switchInfo.getIpAddr(), switchInfo.getUsername(),
+                switchInfo.getPassword());
+        break;
+      }
+      case CISCO_9300: {
+        newController = new Cisco9300(switchInfo.getIpAddr(), switchInfo.getUsername(),
+            switchInfo.getPassword());
+        break;
+      }
+      case OVS_SWITCH: {
+        newController = new OpenVSwitch();
+        break;
+      }
+      default:
+        throw new IllegalArgumentException("Unrecognized switch model "
+            + switchInfo.getModel());
+    }
+    newController.start();
```

Comment:
nit -- start should not be in create.  They're typically two separate phases... first you create but don't really "do" anything, then there's a start after. This allows, for example, changing options or other things before actually activating it. It's fine as is, just something to ponder. Conceptually, the "create" function is the same as the class constructor, only difference is it redirects off of the switch type (aka the factory pattern).